pandas/tests/test_groupby.py

```diff
@@ -3492,6 +3492,24 @@ def test_column_select_via_attr(self):
         expected = self.df.groupby('A').agg(np.mean)
         assert_frame_equal(result, expected)
 
+    def test_rank(self):
+        # GH 11759
+        df = DataFrame({'a': ['A1', 'A1', 'A1'],
+                        'b': ['B1', 'B1', 'B2'],
+                        'c': 1.})
+        df = df.set_index('a')
+        dg = df.groupby('c')
```

Comment:
Done

I'm raising a ValueError for `numeric_only=False`, to parallel what is raised for a non-grouped dataframe, e.g.  `df.rank(method='first')` here.